serverless/networking/serverless-ossm-tls.adoc

```diff
@@ -0,0 +1,168 @@
+
+Example commands should be adjusted according to your domain, sub-domain and CA.
+====
+
+.Procedure
+
+. Create a root certificate and private key to sign the certificates for your services:
++
+[source,terminal]
+----
+$ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=Example Inc./CN=example.com' -keyout example.com.key -out example.com.crt
+----
+. Create a certificate signing request for your domain:
++
+[source,terminal]
+----
+$ openssl req -out custom.example.com.csr -newkey rsa:2048 -nodes -keyout custom.example.com.key -subj "/CN=custom-ksvc-domain.example.com/O=Example Inc."
+----
+. Sign the request with your CA:
++
+[source,terminal]
+----
+$ openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in custom.example.com.csr -out custom.example.com.crt
+----
+. Check that the certificates appear in your directory:
++
+[source,terminal]
+----
+$ ls -1
+----
++
+.Example output
+[source,terminal]
+----
+custom.example.com.crt
+custom.example.com.csr
+custom.example.com.key
+example.com.crt
+example.com.key
+----
+. Create a secret:
++
+[source,terminal]
+----
+$  oc create -n istio-system secret tls custom.example.com --key=custom.example.com.key --cert=custom.example.com.crt
+----
+. Attach the secret to the Istio ingress gateway, by editing the `ServiceMeshControlPlane` resource.
+.. Edit the `ServiceMeshControlPlane` resource:
++
+[source,terminal]
+----
+$ oc edit -n istio-system ServiceMeshControlPlane <control_plane_name>
+----
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+[source,yaml]
+----
+istio-ingressgateway:
+  secretVolumes:
+  - mountPath: /custom.example.com
+    name: custom-example-com
+    secretName: custom.example.com
+----
+. Update the Istio ingress gateway to use your secret.
+.. Edit the `default-gateway` resource:
++
+[source,terminal]
+----
+$ oc edit gateway default-gateway
+----
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+[source,yaml]
+----
+- hosts:
+  - custom-ksvc-domain.example.com
+  port:
+    name: https
+    number: 443
+    protocol: HTTPS
+  tls:
+    mode: SIMPLE
+    privateKey: /custom.example.com/tls.key
+    serverCertificate: /custom.example.com/tls.crt
+----
+. Update the route to use pass-through TLS and `8443` as the `spec.port.targetPort`.
+.. Edit the route:
++
+[source,terminal]
+----
+$ oc edit route -n istio-system hello
+----
+.. Add the following configuration to the route:
++
+[source,yaml]
+----
+spec:
+  host: custom-ksvc-domain.example.com
+  port:
+    targetPort: 8443
+  tls:
+    insecureEdgeTerminationPolicy: None
+    termination: passthrough
+  to:
+    kind: Service
+    name: istio-ingressgateway
+    weight: 100
+  wildcardPolicy: None
+----
+
+.Verification steps
+
+* Access your serverless application by a secure connection that is now trusted by the CA:
++
+[source,terminal]
+----
+$ curl --cacert example.com.crt \
```

Comment:
Couldn't connect to this service using RHSM 2.0.0-2 (control plane version v2.0 or v1.1) on OCP 4.6 at Azure - cc @nak3:

```
$ curl --verbose --cacert example.com.crt \
  --connect-to "custom-ksvc-domain.example.com:443:${ROUTER_INGRESS}:443" \
  https://custom-ksvc-domain.example.com
* Connecting to hostname: 52.185.66.252
* Connecting to port: 443
*   Trying 52.185.66.252:443...
* Connected to 52.185.66.252 (52.185.66.252) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: example.com.crt
  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to custom-ksvc-domain.example.com:443 
* Closing connection 0
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to custom-ksvc-domain.example.com:443
```

I also added the following to SMCP, but didn't help:

```
spec:
  security:
    dataPlane:
      mtls: true
```